sw/device/lib/testing/rv_plic_testutils.c

```diff
@@ -14,10 +14,11 @@ void rv_plic_testutils_irq_range_enable(dif_rv_plic_t *plic,
                                         dif_rv_plic_target_t target,
                                         dif_rv_plic_irq_id_t start_irq_id,
                                         dif_rv_plic_irq_id_t end_irq_id) {
+  rand_testutils_rng_t rand_testutils_rng = rand_testutils_init(NULL);
```

Comment:
`rand_testutils` now provides a context that needs to be maintained _somewhere_. 
It's not a good idea for an another testutil to create this context for itself - we may end up with multiple rand test utils contexts that are separately maintained. 
What we need here is the equivalent of a singleton instance of the context that all requestors use. 

Is having an `extern rand_testutils_rng_t the_one_and_only_rng_ctx` declared in `rand_testutils.h` and having an instance of that created and maintained in `ottf_main.c` (or somewhere more appropriate common testing lib code) the right way to do it? 

@timothytrippel

Reply:
Yeah I think that's the way to go here if you want to ensure only one context is created. Putting it in the OTTF makes perfect sense to me.

Reply:
Done - PTAL, thanks!

Reply:
@timothytrippel how do you think this context needs to be maintained relative to the `entropy_testutil`? 
Since this flow fetches entropy through the `ibex` rand registers, it also means the entropy complex needs to be setup.
Should we just rely on the test_rom setup for that? That might be sufficient...

Reply:
Maybe I should read from entropy source directly rather than from rv_core_ibex?

Reply:
Yeah I was thinking just rely on the test ROM. If we do go that route, then you will have to read from the rv_core_ibex registers I believe since the test ROM [does not](https://cs.opensource.google/opentitan/opentitan/+/master:sw/device/lib/testing/test_rom/test_rom_start.S;drc=b602e150472bd499651e2e6b1ddbfcbb4d852232;l=178) configure the entroy_src to allow for reading from the entropy_data register.

Reply:
Sounds good, I will keep it the way it is. The test ROM does bare-bones entropy complex init, which is good enough for it to work. For tests that do configure the entropy source differently (and also need SW randomization), we could seed the SW PRNG LFSR at the start and then set the `reseed_frequency` to `UINT32_MAX`, so it no longer fetches random data from the entropy SS (via rv_core_ibex), leaving the test undisturbed.